src/stream/stream_backend.rs

```diff
@@ -220,6 +220,75 @@ fn create_udp_stream(
     )));
 }
 
+fn create_rtsp_stream(
+    video_and_stream_information: &VideoAndStreamInformation,
+) -> Result<StreamType, SimpleError> {
+    let encode = video_and_stream_information
+        .stream_information
+        .configuration
+        .encode
+        .clone();
+    let endpoints = &video_and_stream_information.stream_information.endpoints;
```

Comment:
We should accept a single endpoint for the RTSP server, the reason behind is that RTSP has a single endpoint for factory.
We should also accept only endpoints with the format: `rtsp://localhost/path`, where path is not nested. There is no option to provide RTSP servers outside localhost, what we could accept a port for the RTSP server, but we need to make sure that the same port is used between RTSP streams.

Reply:
Great, thanks for pointing that out!

> We should accept a single endpoint for the RTSP server, the reason behind is that RTSP has a single endpoint for factory.

For now I'll get only the first endpoint from the endpoints vector, could it be done in some better way?

---

> We should also accept only endpoints with the format: `rtsp://localhost/path`, where path is not nested. There is no option to provide RTSP servers outside localhost, what we could accept a port for the RTSP server, but we need to make sure that the same port is used between RTSP streams.

Hmm, to solve that, we could:
- check for rtsp scheme
- check if the domain is localhost
- check if there is no more than 1 path_segment (like segmentA/segmentB/segmengC)

Like so:
```rust
    let endpoint = &video_and_stream_information.stream_information.endpoints[0];
if endpoint.scheme() != "rtsp" {
        return Err(SimpleError::new(format!(
            "The URL's scheme for RTSP endpoints should be \"rtsp\", but was: {:?}",
            endpoint.scheme()
        )));
    } else if endpoint.host_str() != "localhost".into() {
        return Err(SimpleError::new(format!(
            "The URL's host for RTSP endpoints should be \"localhost\", but was: {:?}",
            endpoint.host_str()
        )));
    } else if endpoint.path_segments().iter().count() > 1 {
        return Err(SimpleError::new(format!(
            "The URL's path for RTSP endpoints shouldn't have nested segments (e.g.: \"segmentA/segmentB/segmengC\"), but was: {:?}",
            endpoint.path()
        )));
    }
```

Is there a better topology?

---

Lastly, about the `port`, I'm not sure if we can do it, but maybe we can initialize the RTSP server with the port passed in the first call to create_rtsp_stream, then check if it is the same in the subsequent create_rtsp_stream calls, something like:
```rust
    } else if RTSP_SERVER.is_running() && endpoint.port() != RTSP_SERVER.port() {
        return Err(SimpleError::new(format!(
            "The URL's port for RTSP endpoints should all match to {:?}, but was: {:?}",
            RTSP_SERVER.port(),
            endpoint.port()
        )));
    }
```